src/server.rs

```diff
@@ -1,23 +1,50 @@
 use std::collections::HashMap;
+use std::fmt;
 use std::rc::Rc;
 use std::sync::{Arc, Mutex};
 
-use flux::semantic::nodes::FunctionParameter;
+use flux::analyze;
+use flux::ast::walk::walk_rc;
+use flux::ast::walk::Node as AstNode;
+use flux::ast::{Expression, Package, PropertyKey, SourceLocation};
+use flux::parser::parse_string;
+use flux::semantic::nodes::Expression as SemanticExpression;
+use flux::semantic::nodes::{CallExpr, FunctionParameter};
+use flux::semantic::types::{MonoType, Record};
 use flux::semantic::walk;
-use log::{debug, error, info, warn};
-use lspower::jsonrpc::Result;
-use lspower::lsp;
-use lspower::LanguageServer;
+use flux::semantic::walk::Node as SemanticNode;
+use flux::semantic::walk::Visitor as SemanticVisitor;
+use flux::{imports, prelude};
 
 use crate::convert;
 use crate::handlers::document_symbol::sort_symbols;
 use crate::handlers::signature_help::find_stdlib_signatures;
-use crate::handlers::{find_node, NodeFinderResult};
+use crate::shared::ast::is_in_node;
+use crate::shared::get_package_name;
+use crate::shared::signatures::get_argument_names;
+use crate::shared::structs::Function;
+use crate::stdlib::{
+    create_function_signature, get_builtin_functions,
+    get_package_functions, get_package_infos,
+};
+use crate::visitors::ast::package_finder::{
+    PackageFinderVisitor, PackageInfo,
+};
+use crate::visitors::ast::{CallFinderVisitor, NodeFinderVisitor};
+use crate::visitors::semantic::NodeFinderVisitor as SemanticNodeFinderVisitor;
 use crate::visitors::semantic::{
-    DefinitionFinderVisitor, FoldFinderVisitor, IdentFinderVisitor,
-    NodeFinderVisitor, SymbolsVisitor,
+    DefinitionFinderVisitor, FoldFinderVisitor,
+    FunctionFinderVisitor, IdentFinderVisitor, Import,
+    ImportFinderVisitor, ObjectFunctionFinderVisitor, SymbolsVisitor,
 };
 
+use log::{debug, error, info, warn};
```

Comment:
Normally I wouldn't comment on this, but this patch is a bit big. I'd like to avoid a refactor/reorder of `use` statements here. It's hard to see what got imported new and what was here and just got moved.

Reply:
Gotcha. Didn't even realized that I had moved this. Fixing now.